src/octoprint/static/js/app/viewmodels/files.js

```diff
@@ -454,35 +454,44 @@ $(function() {
                     }
                     if(warn)
                     {
-                        var warning = "<p>" + gettext("Revise file ")+ data["name"] + "</p>";
+                        var warning = "<p>" + _.sprintf(gettext("Revise file %s"), data["name"]) + "</p>";
```

Comment:
Ah... yeah... I should have taken a look into that commit first. I see you improved the structure, also used sprintf, good idea!

The streamlining by extracting the boundaries of the volume first, THEN calculating if it is exceeded or not still would make things cleaner.

And you really need to fix your indentation levels, bracket positioning and white space use. Take a look at how the code looks:

![image](https://cloud.githubusercontent.com/assets/83657/14420082/44288912-ffca-11e5-85aa-af7b09701811.png)

That seriously breaks readability :/ Good white space placement is just as important as that the code functions in the first place, because if it takes you minutes to even understand the structure of a piece of code every time you come across it, maintaining it becomes a chore. And code that is hard to maintain is code that quickly becomes buggy and/or messy in other ways. Always imagine that you'll need to open up the file you are working on in a year, with a headache and under time pressure and figure out why something in there doesn't work as expected. And then code so that this future self of you will not want to kill current you with a baseball bat in retrospect.